File: src/main/java/hacker/solution/MyMath.java
package hacker.solution;

public class MyMath {
    public static boolean checker(PerformOperation po, int num) {
        return po.check(num);
    }
    public static PerformOperation isOdd() {
        return n -> (n % 2) != 0;
    }
    public static PerformOperation isPrime() {
        return n -> {
            boolean flag = true;
            for (int i = 2; i <= n / 2; ++i) {
                if (n % i == 0) {
                    flag = false;
                    break;
                }
            }
            return flag;
        };
    }
    public static PerformOperation isPalindrome() {
        return n -> String.valueOf(n).equals(new StringBuffer(String.valueOf(n)).reverse().toString());
    }
}

File: src/main/kotlin/hacker/solutionk/MyMath.kt
package hacker.solutionk

import java.math.BigInteger

object MyMath: PerformOperation {
    override fun isOdd(n: Int): Boolean = mod(n,2) != 0
    override fun isPrime(n: Int): Boolean {
            var flag = true
            for (i in 2..n / 2) {
                if (n % i == 0) {
                    flag = false
                    break
                }
            }
            return flag
    }
    override fun isPalindrome(n: Int): Boolean {
        return n.toString() == StringBuffer(n.toString()).reverse().toString()
    }
    private fun mod(n:Int, i:Int):Int {
       val bn = n.toBigInteger()
       val bi = i.toBigInteger()
       return bn.mod(bi).toInt()
    }
}